scripts/cli/CodeCompass.py

```diff
@@ -0,0 +1,210 @@
+import argparse
+import getpass
+import os
+import sys
+import resource
+
+import handler.parse
+import handler.server
+import handler.workspace
+from util.util import *
+
+#--- Argument parser ---#
+
+parser = argparse.ArgumentParser()
+subparsers = parser.add_subparsers()
+
+#--- Parse handling options ---#
+
+parse_parser = subparsers.add_parser('parse', help = 'Parser handling')
+
+parse_parser.add_argument(
+  '-w', '--workdir', type = str, required = True,
+  help = 'Path of workspace directory of the projects in which config files are\
+  stored.')
+parse_parser.add_argument(
+  '-n', '--name', type = str, required = True,
+  help = 'Project name to identify the project.')
+parse_options = parse_parser.add_mutually_exclusive_group()
+parse_options.add_argument(
+  '-b', '--build', type = str,
+  help = 'This option is used to build the project and create build logs. This\
+  flag has to be given the build command which builds the project.')
+parse_options.add_argument(
+  '-l', '--logfile', type = str,
+  help = 'Path of the parse log which is used by the parser as input.')
+parse_options.add_argument(
+  '-o', '--logonly', type = str,
+  help = 'Log only. In this case parser is not started after logging session.')
+parse_parser.add_argument(
+  '-f', '--force', action = 'store_true',
+  help = 'If -l is given and the log file already exists or -b is given and the\
```

Comment:
```python
  if not os.path.exists(projectdir):
    try:
      os.makedirs(projectdir) # It will create the project directory if it doesn't exist
    except OSError:
      sys.stderr.write('Permission denied to create ' + projectdir + '\n')
      sys.exit(1)
  elif not args.force:
    sys.stderr.write(projectdir + ' already exists. Use -f to force reparsing!\n')
    sys.exit(1)
  else: # The project directory is exists here
    if args.logfile:
```